hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java

```diff
@@ -41,6 +55,19 @@ public DiscoveryJoiner(Node node, DiscoveryService discoveryService, boolean use
     }
 
     @Override
+    protected Collection<Address> getPossibleAddressesForInitialJoin() {
```

Comment:
Shouldn't this be responsibility of `discoveryService` itself? Not the `DiscoveryJoiner`.

Reply:
unfortunately discovery service has no knowledge where the `discoverNodes()` is called when starting a member (and thus the lookup should be more aggressive) or just a regular search (when already joined)

Reply:
I see.